src/etc/inc/util.inc

```diff
@@ -1932,19 +1932,30 @@ function mac_format($clientmac) {
 	}
 }
 
-function resolve_retry($hostname, $retries = 5) {
-
-	if (is_ipaddr($hostname)) {
-		return $hostname;
-	}
+function resolve_retry($hostname, $protocol = 'any', $retries = 5) {
```

Comment:
The protocol defaults to `any` here but there is no special handling for that. `dns_get_record()` could be passed `DNS_A + DNS_AAAA` and it should return whatever it finds in that case. The code below should probably allow that case, and then it wouldn't need to care whether the returned value was IPv4 or IPv6 (it can use `is_ipaddr()` and not the v4/v6 variants).

Safer assumptions for `$protocol` would be to use the explicitly value if set (`any`, `inet`, `inet6`) and perhaps assume `inet` if it's passed an empty value.

Reply:
fixed